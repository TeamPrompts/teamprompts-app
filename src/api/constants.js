export const ACTIVE = 'active';
export const ASC = 'asc';
export const AUTHOR = 'Author';
export const DESC = 'desc';
export const EXAMPLES = 'Examples';
export const FITB = 'FITB';
export const FITBS = 'FITBs';
export const ID = 'ID';
export const NAME = 'Name';
export const PROMPTS = 'Prompts';
export const STATUS = 'Status';
export const TAGS = 'Tags';
