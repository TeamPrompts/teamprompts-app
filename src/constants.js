// XS (4), S (8), M (16), L (32)

export const BLANK = '________________'; // M
export const BLANK_XS = '____';
export const MASTER_DETAIL_V1 = 'MASTER_DETAIL_V1';
export const NONE = 'NONE';
export const SEPARATOR = '__BLANK__';
export const TEAMPROMPTS_APP_EXPERIMENT = 'TEAMPROMPTS_APP_EXPERIMENT';

export const modes = {
  blanks: 'blanks',
  blanksXS: 'blanksXS',
  examples: 'examples',
  input: 'input',
  prompts: 'prompts'
};

export const pageTypes = {
  CategoryPage: 'CategoryPage',
  HomePage: 'HomePage',
  PromptsPage: 'PromptsPage'
};
