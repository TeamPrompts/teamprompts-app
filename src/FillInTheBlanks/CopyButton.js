import React from 'react';
import { BLANK, modes } from '../constants';
import compose from './compose';
import getValues from './getValues';

function makeBuild({ inputs, mode, values }) {
  function build({ index }) {
    let element;
    if (mode === modes.input) {
      if (inputs && inputs[index]) {
        element = inputs[index];
      } else {
        element = BLANK;
      }
    } else {
      element = values[index];
    }
    return element;
  }
  return build;
}

// TODO: read inputs from the local storage
function CopyButton({ fitb, inputs, mode, onClick }) {
  return (
    <button
      className="bg-blue-500 focus:outline-none font-bold font-mono hover:bg-blue-700 ml-3 px-4 rounded-full text-xs text-white"
      data-testid="copy-button"
      onClick={event => {
        event.stopPropagation();
        const values = getValues(mode, fitb);
        const result = compose({
          build: makeBuild({ inputs, mode, values }),
          source: fitb.source,
          values
        });
        onClick(result.join(''));
      }}
      type="button"
    >
      Copy to clipboard
    </button>
  );
}

export default CopyButton;
