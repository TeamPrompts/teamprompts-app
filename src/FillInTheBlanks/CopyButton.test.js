import React from 'react';
import { create } from 'react-test-renderer';
import { modes } from '../constants';
import CopyButton from './CopyButton';

describe('CopyButton', () => {
  const props = {
    mode: modes.input
  };

  it('to match snapshot', () => {
    const tree = create(<CopyButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
