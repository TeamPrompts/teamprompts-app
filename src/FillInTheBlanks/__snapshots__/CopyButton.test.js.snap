// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyButton to match snapshot 1`] = `
<button
  className="bg-blue-500 font-bold font-mono hover:bg-blue-700 ml-3 px-4 rounded-full text-sm text-white"
  onClick={[Function]}
  type="button"
>
  Ctrl+C
</button>
`;
